source/jormungandr/jormungandr/street_network/here.py

```diff
@@ -318,32 +440,73 @@ def get_matrix_params(self, origins, destinations, mode, max_duration, request):
         datetime = request['datetime']
         params['departure'] = _str_to_dt(datetime)
 
-        # Note: for the moment we limit to self.max_points the number of n:m asked
-        for i, o in enumerate(origins[: self.max_points]):
+        for i, o in enumerate(origins[: int(max_matrix_points)]):
             params['start{}'.format(i)] = _get_coord(o)
 
-        for i, o in enumerate(destinations[: self.max_points]):
+        for i, o in enumerate(destinations[: int(max_matrix_points)]):
             params['destination{}'.format(i)] = _get_coord(o)
 
         # TODO handle max_duration (but the API can only handle a max distance (searchRange)
 
         return params
 
-    def get_street_network_routing_matrix(
-        self, instance, origins, destinations, mode, max_duration, request, request_id, **kwargs
+    def _get_sn_routing_matrix_with_simple_matrix(
+        self, origins, destinations, mode, max_duration, request, realtime_traffic, max_matrix_points
     ):
-        params = self.get_matrix_params(origins, destinations, mode, max_duration, request)
+        params = self.get_matrix_params(
+            origins, destinations, mode, max_duration, request, realtime_traffic, max_matrix_points
+        )
         r = self._call_here(self.matrix_service_url, params=params)
-
         if r.status_code != 200:
             self.log.error(
                 'impossible to query HERE: {} with {} response: {}'.format(r.url, r.status_code, r.text)
             )
             raise TechnicalError('invalid HERE call, impossible to query: {}'.format(r.url))
 
-        resp_json = r.json()
+        return self._create_matrix_response(r.json(), origins, destinations)
 
-        return self._get_matrix(resp_json, origins, destinations)
+    def _get_sn_routing_matrix_with_multi_direct_path(
+        self, origins, destinations, mode, max_duration, request, realtime_traffic, max_matrix_points
+    ):
+        sn_routing_matrix = response_pb2.StreetNetworkRoutingMatrix()
+        row = sn_routing_matrix.rows.add()
+        for origin in origins[: int(max_matrix_points)]:
+            for destination in destinations[: int(max_matrix_points)]:
+                params = self.get_direct_path_params(
+                    origin,
+                    destination,
+                    mode,
+                    PeriodExtremity(datetime=request['datetime'], represents_start=request['clockwise']),
+                    realtime_traffic,
+                )
+                r = self._call_here(self.routing_service_url, params=params)
```

Comment:
If I'm not mistaken, each call is made sequentially.
Is it possible to make them concurrently to speed things up ?

Reply:
yes, this is a step by step calling method. This is not efficient. 
But since we have choose to call the matrix Here API... this method is not used.
I added a TODO for the future to propose the parallelize method.